src/general/world_effects/PhotosynthesisProductionEffect.cs

```diff
@@ -114,27 +113,48 @@ private void ApplyCompoundsAddition()
                     {
                         if (output.Key.ID is Compound.Oxygen)
                         {
-                            oxygenBalance += output.Value * outputModifier * effectiveSpeed * species.Value;
+                            oxygenOut += output.Value * outputModifier * effectiveSpeed * species.Value;
                         }
                         else if (output.Key.ID is Compound.Carbondioxide)
                         {
-                            co2Balance += output.Value * outputModifier * effectiveSpeed * species.Value;
+                            co2Out += output.Value * outputModifier * effectiveSpeed * species.Value;
                         }
                     }
                 }
             }
 
-            // Scale the balances to make the changes less drastic
-            oxygenBalance *= modifier;
-            co2Balance *= modifier;
+            float oxygenTarget;
+            float co2Target;
+
+            patch.Biome.TryGetCompound(Compound.Oxygen, CompoundAmountType.Biome, out var existingOxygen);
+            patch.Biome.TryGetCompound(Compound.Carbondioxide, CompoundAmountType.Biome, out var existingCo2);
+
+            float total = existingOxygen.Ambient + existingCo2.Ambient;
 
-            changesToApply.Clear();
+            if (oxygenOut == 0)
+            {
+                if (co2Out == 0)
+                {
+                    // in the rare event we aren't making either compound, do nothing
+                    continue;
+                }
 
-            if (oxygenBalance != 0)
-                changesToApply[Compound.Oxygen] = oxygenBalance;
+                oxygenTarget = 0;
+                co2Target = total;
+            }
+            else if (co2Out == 0)
+            {
+                oxygenTarget = total;
+                co2Target = 0;
+            }
+            else
+            {
+                oxygenTarget = oxygenOut / (oxygenIn + co2In) * total;
```

Comment:
Could you add a comment here explaining the logic? I don't really understand this kind of scaling because ApplyLongTermCompoundChanges *already* does a relative applying of gases. So higher oxygen output reduces the effect of the co2 adding and vice versa. So to me it looks like this is now doing the same effect twice, which doesn't logically make a ton of sense as ApplyLongTermCompoundChanges needs to be given the absolute change wanted and then it automatically makes gases related to each other and caps it to a 100%.

Reply:
Fair. Rather than explain here, I will add the comment and see if the explanation makes sense to you.